include/core/Layer.hpp

```diff
@@ -128,6 +132,10 @@ namespace ngen
                              +" at feature id "+id;
                     throw models::external::State_Exception(msg);
                 }
+#if NGEN_WITH_ROUTING
+                int results_index = catchment_indexes[id];
+                catchment_results[results_index + current_step] = response;
```

Comment:
Because we eventually need to be able to feed the data for each step as a contiguous array of the collected catchments' results into t-route, I'd like to transpose this:
```suggestion
                catchment_results[current_step * catchment_indexes.size() + catchment_indexes[id]] = response;
```
With the index calculation below becoming `catchment_indexes[feature_id] = i`, without multiplication by `num_times` there.

Same applies for nexuses, too

Reply:
Oh, actually, I had a better thought on this, before too much gets revised. Change the parameters from `vector` to `span`, and then at the call sites, pass `span(results.data() + current_step * feature_index.size()`, then the subscript here will just be `feature_index[id]`